plugins/filter_aws/aws.c

```diff
@@ -466,6 +570,14 @@ static int get_ec2_metadata(struct flb_filter_aws *ctx)
         ctx->new_keys++;
     }
 
+    if (ctx->tags_include && (!ctx->tag_keys || !ctx->tag_values)) {
```

Comment:
since you have the `tags_fetched` variable, I think that should be used in the check here to determine if we've fetched them or not.